Remove deprecated config options

mkorszun opened the pull request:
* Remove support for `uri` config option - replaced by `connection`
* Remove support for `queue_ttl` - replaced by `queue_options.arguments`
* Remove support for `queue_max_priority` - replaced by `queue_options.arguments`
* Remove all unsupported config options from guides, examples and documentation

Issues addressed:
* https://github.com/meltwater/gen_rmq/issues/220
* https://github.com/meltwater/gen_rmq/issues/219
* https://github.com/meltwater/gen_rmq/issues/218

## Description
<!--- Describe your changes in detail -->
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
<!--- Include examples on how to use the functionality if applicable -->
<!--- Describe any manual or special tests you have done -->
<!--- Attach screenshots if appropriate -->

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- Ignore the ones that are not applicable -->
- [ ] I have added unit tests to cover my changes.
- [ ] I have improved the code quality of this repo. (refactoring, or reduced number of static analyser issues)
- [ ] I have updated the documentation accordingly

spier commented:
@mkorszun were the 3 config options above introduced in v3.0?

I am wondering where we should document such changes. Maybe in the [migration guide to 3.0](https://github.com/meltwater/gen_rmq/blob/master/documentation/migrations/3.0.0.md)?

mkorszun commented:
@spier no, we have introduced them before v3.0. When we introduced them, it was not a breaking change. Generated documentation + a couple of guides was enough - we believed. Now it is a time to remove deprecated ones.

akoutmos commented:
Looks good to me. We can ignore the dialyzer error for now. Opened an issue and Jose provided a fix https://github.com/elixir-lang/elixir/issues/10460

mkorszun commented:
@Shemeikka I have discovered one thing: max priority trimming (to 255) was happening only if specified with deprecated config option - `queue_max_priority`. If part of the queue arguments, no trimming. Is it desired? 

Would you mind having another round of review? I have removed other parts of code that looked to be dead / not used.

Shemeikka commented:
> @Shemeikka I have discovered one thing: max priority trimming (to 255) was happening only if specified with deprecated config option - `queue_max_priority`. If part of the queue arguments, no trimming. Is it desired?
> 

Good catch! I forgot about this and missed it when I reviewed the changes.

It's not desired. The max priority should be between 1 and 255.

Shemeikka commented on lib/gen_rmq/consumer/queue_configuration.ex:
```diff
@@ -162,11 +81,4 @@ defmodule GenRMQ.Consumer.QueueConfiguration do
     end
   end
-
-  defp set_max_priority_to_highest_value(mp)
-       when is_integer(mp) and mp > @max_priority do
-    255
-  end
-
-  defp set_max_priority_to_highest_value(mp) when is_integer(mp), do: mp
```
These functions should be kept and used when setting the max priority using queue arguments

Shemeikka commented on lib/gen_rmq/consumer/queue_configuration.ex:
```diff
@@ -102,35 +46,10 @@ defmodule GenRMQ.Consumer.QueueConfiguration do
-    end
-  end
-
-  defp setup_priority(options) do
-    case options[:max_priority] do
-      nil ->
-        options
-
-      value ->
-        value = set_max_priority_to_highest_value(value)
-        args = Keyword.get(options, :arguments, [])
-        Keyword.put(options, :arguments, [{"x-max-priority", :long, value} | args])
-    end
-  end
```
Something similar is needed since the x-max-priority argument value should be between 1 and 255, so we should call the 'set_max_priority_to_highest_value' to check if the value is above the limit.

mkorszun replied:
all right, will do smth similar 👍

mkorszun replied:
@Shemeikka should be fixed in the latest commit -> https://github.com/meltwater/gen_rmq/pull/221/commits/36b205e7f294e2dc899f32d8625b49af670ebbc9

Shemeikka replied:
Looks good

Shemeikka reviewed (changes requested):
Checked the code again and added comments to the removed functionality that is still required.

Shemeikka commented:
@mkorszun Is there any work left on this? It seems that this should now be working.

mkorszun commented:
@Shemeikka not really, just lost it from my radar. Merging it now
